src/components/DayPicker.jsx

```diff
@@ -174,21 +204,107 @@ export default class DayPicker extends React.Component {
         this.adjustDayPickerHeight();
       }
     }
+
+    if (isFocused !== this.props.isFocused) {
+      if (isFocused) {
+        const focusedDate = this.getFocusedDay(currentMonth);
+        this.setState({
+          showKeyboardShortcuts,
+          focusedDate,
+        });
+      } else {
+        this.setState({ focusedDate: null });
+      }
+    }
   }
 
   shouldComponentUpdate(nextProps, nextState) {
     return shallowCompare(this, nextProps, nextState);
   }
 
   componentDidUpdate(prevProps, prevState) {
-    if (this.state.monthTransition || !this.state.currentMonth.isSame(prevState.currentMonth)) {
+    const { monthTransition, currentMonth } = this.state;
+    if (monthTransition || !currentMonth.isSame(prevState.currentMonth)) {
       if (this.isHorizontal()) {
         this.adjustDayPickerHeight();
       }
     }
+
+    if (!prevProps.showKeyboardShortcuts && this.props.showKeyboardShortcuts) {
+      // not sure what the right approach is here
+      this.container.focus();
+    }
+  }
+
+  onKeyDown(e) {
+    const { onBlur } = this.props;
+    const { focusedDate } = this.state;
+
+    if (!focusedDate) return;
+
+    const newFocusedDate = focusedDate.clone();
+
+    let didTransitionMonth = false;
+
+    switch (e.key) {
+      case 'ArrowUp':
+        newFocusedDate.subtract(1, 'week');
+        didTransitionMonth = this.maybeTransitionPrevMonth(newFocusedDate);
+        break;
+      case 'ArrowLeft':
+        newFocusedDate.subtract(1, 'day');
+        didTransitionMonth = this.maybeTransitionPrevMonth(newFocusedDate);
+        break;
+      case 'Home':
+        newFocusedDate.startOf('week');
+        didTransitionMonth = this.maybeTransitionPrevMonth(newFocusedDate);
+        break;
+      case 'PageUp':
+        newFocusedDate.subtract(1, 'month');
+        didTransitionMonth = this.maybeTransitionPrevMonth(newFocusedDate);
+        break;
+
+      case 'ArrowDown':
+        newFocusedDate.add(1, 'week');
+        didTransitionMonth = this.maybeTransitionNextMonth(newFocusedDate);
+        break;
+      case 'ArrowRight':
+        newFocusedDate.add(1, 'day');
+        didTransitionMonth = this.maybeTransitionNextMonth(newFocusedDate);
+        break;
+      case 'End':
+        newFocusedDate.endOf('week');
+        didTransitionMonth = this.maybeTransitionNextMonth(newFocusedDate);
+        break;
+      case 'PageDown':
+        newFocusedDate.add(1, 'month');
+        didTransitionMonth = this.maybeTransitionNextMonth(newFocusedDate);
+        break;
+
+      case '?':
+        this.toggleKeyboardShortcuts();
```

Comment:
This is probably nitpicking, but can there be an easy way to detect if the shortcut panel was activated by DateInput, hence if we're closing the shortcut panel it would focus back on the DateInput? (I tried played with it but lost myself in managing props)